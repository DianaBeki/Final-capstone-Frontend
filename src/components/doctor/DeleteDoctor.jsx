import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchDoctors, deleteDoctor } from '../../redux/doctor/doctorSlice';

const DeleteDoctor = () => {
  const { doctors, status, error } = useSelector((store) => store.doctor);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchDoctors());
  }, [dispatch]);
  const handleDelete = (id) => {
    dispatch(deleteDoctor(id));
  };
  return (
    <>
      <div className="text-center">
        <h1 className="mb-2 font-bold  text-[14px] md:text-[25px] text-[#1a1a1a] md:mb-5">Delete Doctor</h1>
      </div>
      {status === 'loading' && <p>Loading...</p>}
      {status === 'failed' && (
        <p>
          Error:
          {error}
        </p>
      )}
      {doctors.length === 0 ? (<h1 className="text-xl mt-20 text-slate-500"> No doctors Added!!</h1>)
        : (doctors.map((doctor) => (
          <div key={doctor.id} className="flex flex-col justify-center items-center py-[.8rem] md:gap-5 md:grid md:grid-cols-2 md:px-[8rem] md:py-[1.2rem]">
            <img src={doctor.image} alt={doctor.name} className="object-cover w-83" />
            <div>
              <div>
                <p className="font-bold text-[20px]">{doctor.name}</p>
                <p className="text-[10px] text-gray-500 font-semi-bold md:text-[18px] md:leading-[30px]">{doctor.specialization}</p>
              </div>
              <button type="button" className="bg-red-700 hover:bg-gray-400 text-white px-1.5 md:px-4 md:py-2 rounded-md transition-transform transform hover:scale-105  duration-500" onClick={() => { handleDelete(doctor.id); }}>
                Delete
              </button>
            </div>
          </div>
        )))}
    </>
  );
};

export default DeleteDoctor;
