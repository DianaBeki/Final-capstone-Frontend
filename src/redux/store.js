import { configureStore } from '@reduxjs/toolkit';
import doctorReducer from './doctor/doctorSlice';
import userSlice from './user/userSlice';

const store = configureStore({
  reducer: {
    doctor: doctorReducer,
    user: userSlice,
  },
});

export default store;
