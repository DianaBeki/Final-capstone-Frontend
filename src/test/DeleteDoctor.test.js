import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import DeleteDoctor from '../components/doctor/DeleteDoctor';
import store from '../redux/store';

const render = (component) => rtlRender(
  <Provider store={store}>
    {component}
  </Provider>,
);

describe('DeleteDoctor Component', () => {
  it('renders DeleteDoctor component', () => {
    const { container } = render(<DeleteDoctor />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
