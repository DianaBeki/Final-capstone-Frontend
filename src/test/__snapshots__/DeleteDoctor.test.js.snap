// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteDoctor Component renders DeleteDoctor component 1`] = `
<div
  class="text-center"
>
  <h1
    class="font-bold text-[#1a1a1a] mb-5"
  >
    Delete Doctor
  </h1>
</div>
`;
